docs/architecture/adr-030-rosetta-api-support.md

```diff
@@ -0,0 +1,223 @@
+
+## Decision
+
+
+The service is structured in a way that:
+
+1. It becomes easy to inject different implementations for different types of SDK. For this abstraction we have used the term Adapter.
+2. Due to the nature of versioning that has been done with Cosmos SDK so far it becomes very difficult to have different SDK versions included in the same repo, for that we created a generic shared code that includes the Service and the Adapter interface.
+3. It is easy to inject and instantiate wherever is needed in the different applications.
+
+
+## Architecture
+
+
+### Service
+
+`Service` is a simple `struct` that is started and listens to the port specified in the options. This is meant to be used across all the Cosmos SDK versions that are actively supported.
+
+The constructor follows:
+
+`func New(options Options, network Network) (*Service, error)`
+
+It accepts an `Options` `struct` that holds service configuration values, such as the port the service would be listening to:
+
+```
+type Options struct {
+	Port uint32
+}
+```
+
+The `Network` type holds network-specific properties (i.e. configuration values) and adapters. Ready-to-use Cosmos SDK release series-specific base `Network` concrete types will be made available. Applications can embed such types in their own custom types.
+
+```
+type Network struct {
+	Properties rosetta.NetworkProperties
+	Adapter    rosetta.Adapter
+}
+````
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service` to run:
+
+```
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	AddrPrefix          string
+	SupportedOperations []string
+}
+```
+
+Rosetta API services use Blockchain and Network as identifiers, e.g. the developers of gaia, the application that powers the Cosmos Hub, may want to set those to Cosmos Hub and cosmos-hub-3 respectively.
+
+`AddrPrefix` contains the network-specific address prefix. Cosmos SDK base implementations will default to `cosmos`, client applications are instructed that this should be changed according to their network configuration.
+
+`SupportedOperations` contains the transaction types that are supported by the library. At the present time, only `Transfer` is supported.
+
+`Network` holds an `Adapter` reference too. Adapter implementations may vary across different Cosmos SDK release series:
+
+```
+type Adapter interface {
+	DataAPI
+	ConstructionAPI
+}
+
+type DataAPI interface {
+	server.NetworkAPIServicer
+	server.AccountAPIServicer
+	server.MempoolAPIServicer
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+
+````
+
+### Cosmos SDK Integration
+
+The best way to do it would be that every Cosmos SDK version holds the implementation of the Network struct, this can be an example located in the Cosmos SDK Launchpad version.
```

Comment:
```suggestion
Cosmos SDK provides a base `Network` struct so that it could serve as code example for client application developers and testing tool to be used in conjuction with `simd`.
```